### PR TITLE
r.extract: new addon for extracting categories 

### PR DESCRIPTION
**Pull request by petrasovaa**
This is a wrapper around r.reclass to simplify extraction of categories in integer raster maps:
```
r.extract input=zipcodes output=selected_zipcodes cats=27605,27601 -c
```

### CODE REVIEW

File: grass7/raster/r.extract/r.extract.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<h2>DESCRIPTION</h2>
+
+This module extracts selected categories from integer raster map
```
Comment (neteler) [inline grass7/raster/r.extract/r.extract.html:3]:
```suggestion
This module extracts selected categories from an integer raster map
```

File: grass7/raster/r.extract/r.extract.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
```
Comment (neteler) [inline grass7/raster/r.extract/r.extract.py:2]: Please add the common header with copyright, license etc.

File: grass7/raster/r.extract/r.extract.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+#%module
+#% label: Extracts specified categories of an integer input map
```
Comment (neteler) [inline grass7/raster/r.extract/r.extract.py:4]:
```suggestion
#% description: Extracts specified categories of an integer input map.
```

File: grass7/raster/r.extract/r.extract.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+#% keyword: raster
+#% keyword: extract
+#% keyword: extent
```
Comment (neteler) [inline grass7/raster/r.extract/r.extract.py:7]:
```suggestion
#% keyword: extent
#% keyword: category
```

File: grass7/raster/r.extract/r.extract.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+def main():
+    options, flags = gs.parser()
+    original = options["input"]
+    output = options["output"]
+    cats = options["cats"]
+
+    rules = parse(original, cats)
+    if flags["c"] and flags["s"]:
+        gs.warning(
+            _("The extent of the output reclassified" " raster cannot be changed")
```
Comment (neteler) [inline grass7/raster/r.extract/r.extract.py:81]: Is a variable missing here or just a small formatting glitch?

**Comment by petrasovaa** [comment]
Thanks @neteler for the review!

**Pull request merged**